add apache ignite helm chart

aasiutin opened the pull request.

aasiutin committed:
add apache ignite helm chart

Signed-off-by: Oleksii Asiutin <sutok85@gmail.com>

GreyTeardrop committed:
Disable Ignite persistence if volumes are disabled

Signed-off-by: Mykola Rybak <mykola.rybak@gmail.com>

GreyTeardrop committed:
Expose peerClassLoadingEnabled property

Signed-off-by: Mykola Rybak <mykola.rybak@gmail.com>

GreyTeardrop committed:
Expose environment variable configuration

Signed-off-by: Mykola Rybak <mykola.rybak@gmail.com>

aasiutin committed:
add service account to support rbac

Signed-off-by: Oleksii Asiutin <sutok85@gmail.com>

aasiutin committed:
add ability to disable rbac

Signed-off-by: Oleksii Asiutin <sutok85@gmail.com>

viglesiasce commented on stable/ignite/Chart.yaml:
```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+appVersion: "1.0"
```
From the image tag it looks like this should be 2.6.0... Is that the case or is the image tag version always going to be different from the app version?

aasiutin replied:
I think it's better to delete it, cause we want ignite version to be configurable through `values.yaml` so `appVersion` is unnecessary. What do you think if I delete it?

viglesiasce replied:
appVersion is actually used as metadata for things like hub.kubeapps.com. Its useful to have it be meaningful.

aasiutin replied:
Got it, thanks! I've updated image tag and also changed appVersion to 2.7.0. (the latest one)

viglesiasce commented on stable/ignite/values.yaml:
```diff
@@ -0,0 +1,51 @@
+
+rbac: yes
+
+ignite:
```
Is there a reason for this to be nested? Are there other components?

aasiutin replied:
Everything under `ignite:` is application configurations. Everything top-level is release configurations. We followed this logic. Do you think it's better to put everything under `ignite also` to top-level?

viglesiasce replied:
It caught me a bit off guard since the chart/app and image already reference ignite. Additionally its what ive seen most commonly:
- Multi component apps nest things
- Single component apps have them all top-level

aasiutin replied:
I see, I've moved everything to the top level.

aasiutin committed:
change appVersion and chart version

Signed-off-by: Oleksii Asiutin <sutok85@gmail.com>

aasiutin committed:
move all ignite: configs to top-level

Signed-off-by: Oleksii Asiutin <sutok85@gmail.com>

aasiutin committed:
update app version to 2.7.0

Signed-off-by: Oleksii Asiutin <sutok85@gmail.com>

aasiutin committed:
fixed cluster activation commands

new ignite version (2.7.0) has different binaries path

Signed-off-by: Oleksii Asiutin <sutok85@gmail.com>

viglesiasce commented on stable/ignite/values.yaml:
```diff
@@ -0,0 +1,50 @@
+nameOverride: ""
+fullnameOverride: ""
+
+rbac: yes
```
Sorry I missed this before, the standards for RBAC are as follows:
https://docs.helm.sh/chart_best_practices/#role-based-access-control

aasiutin replied:
Oh, I missed this too :(
I'll review this and make fixes, thanks!

aasiutin replied:
I've edited the chart according to the guide. Could you please check it again?

viglesiasce commented on stable/ignite/values.yaml:
```diff
@@ -0,0 +1,50 @@
+  IGNITE_QUIET: "false"
+  JVM_OPTS: "-Djava.net.preferIPv4Stack=true"
+peerClassLoadingEnabled: false
+dataStorage:
```
In most charts they use `persistence` as the key here:
https://github.com/helm/charts/blob/master/stable/wordpress/values.yaml#L228
https://github.com/helm/charts/blob/master/stable/redis/values.yaml#L190

aasiutin replied:
This is fixed

aasiutin committed:
change "dataStorage" config key to "persistence"

Most helm charts use "persistence" config key. So we follow the
same convention.

Signed-off-by: Oleksii Asiutin <sutok85@gmail.com>

aasiutin committed:
adapt chart configs to RBAC standards

according to this link
https://docs.helm.sh/chart_best_practices/#role-based-access-control

Signed-off-by: Oleksii Asiutin <sutok85@gmail.com>
